types/ibc.go

```diff
@@ -41,18 +41,30 @@ func (t IBCTimeoutBlock) IsZero() bool {
 	return t.Revision == 0 && t.Height == 0
 }
 
+type IBCTimeoutBoth struct {
+	Block     IBCTimeoutBlock `json:"block"`
+	Timestamp uint64          `json:"timestamp_nanos"` // TODO: simplify to just "timestamp" in Rust
+}
+
+// IBCTimeout is the timeout for an IBC packet. At least one of block and timestamp is required.
+//
+// This is an enum, i.e. only one of the three fields must be set.
+type IBCTimeout struct {
+	// Nanoseconds since UNIX epoch
+	// See https://golang.org/pkg/time/#Time.UnixNano
+	Timestamp *uint64 `json:"timestamp_nanos,omitempty"` // TODO: simplify to just "timestamp" in Rust
+	// Block after which the packet times out.
+	Block *IBCTimeoutBlock `json:"block,omitempty"`
+	// Block and time after which the packet times out.
+	Both *IBCTimeoutBoth `json:"both,omitempty"`
+}
+
 type IBCPacket struct {
 	Data     []byte      `json:"data"`
 	Src      IBCEndpoint `json:"src"`
 	Dest     IBCEndpoint `json:"dest"`
 	Sequence uint64      `json:"sequence"`
-	// block after which the packet times out.
-	// at least one of timeout_block, timeout_timestamp is required
-	TimeoutBlock *IBCTimeoutBlock `json:"timeout_block,omitempty"`
-	// Nanoseconds since UNIX epoch
-	// See https://golang.org/pkg/time/#Time.UnixNano
-	// at least one of timeout_block, timeout_timestamp is required
-	TimeoutTimestamp *uint64 `json:"timeout_timestamp,omitempty"`
+	Timeout  IBCTimeout  `json:"timeout"`
```

Comment:
This line introduces a bug because in cosmwasm-std the `IbcPacket` still uses the old format with fields `timeout_block` and `timeout_timestamp`.